tests/integration/components/add-env-var-test.js

```diff
@@ -105,4 +117,51 @@ module('Integration | Component | add env-var', function (hooks) {
     var done = assert.async();
     done();
   });
+
+  test('it adds an env var with branch on submit', async function (assert) {
+    assert.expect(7);
+
+    this.owner.register('transform:boolean', DS.BooleanTransform);
+    var store = this.owner.lookup('service:store');
+    assert.equal(store.peekAll('envVar').get('length'), 0, 'precond: store should be empty');
+
+    var repo;
+    run(function () {
+      repo  = store.push({ data: { id: 1, type: 'repo', attributes: { slug: 'travis-ci/travis-web' } } });
+    });
+
+    this.set('repo', repo);
+
+    let branchName = 'foo';
+
+    server.create('branch', {
+      name: branchName,
+      id: `/v3/repo/${this.repository.id}/branch/food`,
+      exists_on_github: true,
+      repository: this.repository,
+    });
+
+    await render(hbs`{{add-env-var repo=repo}}`);
+
+    await fillInWithKeyEvent('.env-name', 'FOO');
+    await fillInWithKeyEvent('.env-value', 'bar');
+
+    await selectSearch('.env-branch-selector', branchName);
+    await selectChoose('.env-branch-selector', branchName);
+
+    await click('.form-submit');
+
+    assert.equal(store.peekAll('envVar').get('length'), 1, 'env var should be added to store');
+
+    var envVar = store.peekAll('envVar').objectAt(0);
+
+    assert.equal(envVar.get('name'), 'FOO', 'name should be set for the env var');
+    assert.equal(envVar.get('value'), 'bar', 'value should be set for the env var');
+    assert.equal(envVar.get('branch'), 'foo', 'branch should be set for the env var');
+    assert.equal(envVar.get('repo.slug'), 'travis-ci/travis-web', 'repo should be set for the env var');
+    assert.ok(!envVar.get('public'), 'env var should be private');
+
+    var done = assert.async();
+    done();
+  });
```

Comment:
Thanks for adding the integration tests, they are definitely useful! Although, acceptance tests are more important, as they test the entire application at once. Could you please also add assertions to [this acceptance test](https://github.com/travis-ci/travis-web/blob/master/tests/acceptance/repo/settings-test.js#L180)

Reply:
Integration tests now checks for empty and non empty branch in env var.